Switch pane: refine Dimming Slider to design specs

ybott-qinetic opened the pull request:
The left padding is used for On/Off Label, displays current state of the switch.
    
/SwitchableOutput/<x>/Settings/DimmingMax|DimmingMin sets the sliders range.
/SwitchableOutput/<x>/Dimming sets the value.
    
Contributes to #2301

blammit commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,53 +4,134 @@
-		id: handleItem
 		parent: root.maskSource
-		width: Theme.geometry_switch_indicator_width
+		width: 3*Theme.geometry_dimmingSlider_handle_width
```
What's the 3* for? If it's for extra padding around the separator, can you add a comment?

blammit commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,53 +4,134 @@
+	Rectangle {
+		x: Theme.geometry_dimmingSlider_leftPadding - width
+		anchors.verticalCenter: parent.verticalCenter
+		width: Theme.geometry_dimmingSlider_seperator_width
```
Typo `seperator`

ybott-qinetic replied:
Done

blammit commented on components/controls/DimmingSlider.qml:
```diff
@@ -65,7 +146,7 @@ Slider {
+			mouse.accepted = (mouseX < root.leftPadding)
 		}
 	}
 }
```
Currently, clicking anywhere on the slider will toggle the on/off state, but it should only allow that if you click in the far-left button area.

ybott-qinetic replied:
I don't get that behaviour. I will check on device.

blammit replied:
Never mind -  I don't see it with the latest PR anymore 👍

blammit commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,53 +4,134 @@
+		anchors.leftMargin: Theme.geometry_dimmingSlider_label_leftMargin
+		anchors.verticalCenter: parent.verticalCenter
+		text: CommonWords.onOrOff(dimmingState.expectedValue)
+		font.pixelSize: Theme.font_size_body2
```
Font size needs to be reduced?

blammit replied:
Can you centre-align the text horizontally, otherwise the padding is reduced on the right when the text is "Off":

<img width="121" height="96" alt="image" src="https://github.com/user-attachments/assets/9c46c301-7718-4052-96d9-7adcc4ac95a5" />

blammit commented on themes/geometry/SevenInch.json:
```diff
@@ -64,7 +64,13 @@
+    "geometry_dimmingSlider_handle_width": 4,
+    "geometry_dimmingSlider_handle_vertical_padding": 6,
+    "geometry_dimmingSlider_seperator_width": 2,
+    "geometry_dimmingSlider_seperator_vertical_padding": 6,
```
Could just have the one `geometry_dimmingSlider_vertical_padding`, instead of the same value for handle vs separator padding

blammit reviewed (commented):
Looks great, works nicely.

blammit commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,53 +4,136 @@
+		text: root.text
+		horizontalAlignment: Text.AlignHCenter
+		font.pixelSize: Theme.font_size_body1
+		color: Theme.color_white
```
This should use the primary colour instead (which is the default for `Label`).

blammit commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,53 +4,136 @@
+		anchors.verticalCenter: parent.verticalCenter
+		width: Theme.geometry_dimmingSlider_separator_width
+		height: parent.height - Theme.geometry_dimmingSlider_decorator_vertical_padding*2
+		radius: Theme.geometry_dimmingSlider_seperator_width/2
```
Could you add some spaces and parenthesis, like:

```
height: parent.height - (Theme.geometry_dimmingSlider_decorator_vertical_padding * 2)
radius: Theme.geometry_dimmingSlider_seperator_width / 2
```

Similarly, elsewhere in the patch.

blammit commented:
Also as discussed, avoid using "Partially fixes X" in the commit message as that will auto-close the task.

blammit reviewed (commented):
As commented previously, the commit message needs to avoid the "Partially fixes" text.

blammit commented on components/SwitchableOutputDelegate.qml:
```diff
@@ -190,10 +190,11 @@ BaseListItem {
+				: (dimmingState.expectedValue === 1 ? Theme.color_font_disabled : Theme.color_gray3)
+			from: dimmingMin.valid ? dimmingMin.value : 0
+			to: dimmingMax.valid ? dimmingMax.value : 100
+			stepSize: stepsize.valid ? stepsize.value : 1
```
Does the `stepSize` need to be set? `DimmingSlider` does not use it.

ybott-qinetic replied:
No, I have removed it.

ybott-qinetic commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,68 +4,154 @@
+			horizontalAlignment: Text.AlignHCenter
+			verticalAlignment: Text.AlignVCenter
+			font.pixelSize: Theme.font_size_body1
+			color: root.enabled ? Theme.color_font_primary : Theme.color_gray6
```
Changed (I tried a lot of the defined colors for this and settled on white).

ybott-qinetic replied:
Fixed.

ybott-qinetic commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,68 +4,154 @@
+	leftPadding: textMetrics.width + (Theme.geometry_dimmingSlider_text_padding * 2) + Theme.geometry_dimmingSlider_separator_width
+
+	background: Rectangle {
+		width: root.availableWidth + root.leftPadding
```
I have changed this to follow parent.width.

ybott committed:
Switch pane: refine Dimming Slider to design specs

The left padding is used for On/Off Label, displays current state
of the switch.

/SwitchableOutput/<x>/Settings/DimmingMax|DimmingMin sets the sliders
range.
/SwitchableOutput/<x>/Dimming sets the value.

Contributes to #2301

ybott-qinetic commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,68 +4,154 @@
+	TextMetrics {
+		id: textMetrics
+		font.pixelSize: Theme.font_size_body1
+		text: (CommonWords.on.length > CommonWords.off.length) ? CommonWords.on : CommonWords.off
```
Makes sense.

blammit reviewed (approved):
Looks great.

chriadam commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,68 +4,153 @@
-		border.color: Theme.color_ok
-		radius: Theme.geometry_button_radius
+	MouseArea {
+		anchors.left: parent.left
```
prefer grouped property assignment if there are more than one, here and elsewhere for anchors.

ybott-qinetic replied:
Changed.

chriadam commented on components/controls/DimmingSlider.qml:
```diff
@@ -4,68 +4,153 @@
+
+	// The Slider padding area is used to display the dimming slider state and provide
+	// click handling for the control as well as the inherited slider behaviour.
+	leftPadding: textMetrics.width + (Theme.geometry_dimmingSlider_text_padding * 2) + Theme.geometry_dimmingSlider_separator_width
```
see https://bugreports.qt.io/browse/QTBUG-124588 for in-depth discussion, but basically you need both the bounding rect width plus bounding rect x, not just width.

ybott-qinetic replied:
Fixed.
